crates/cac_client/src/interface.rs

```diff
@@ -127,10 +130,7 @@ pub extern "C" fn cac_start_polling_update(tenant: *const c_char) {
         null_check!(client, "CAC client for tenant not found", return);
         let local = task::LocalSet::new();
         // println!("in FFI polling");
-        local.block_on(
-            &Runtime::new().unwrap(),
-            (*client).clone().run_polling_updates(),
-        );
+        local.block_on(&CAC_RUNTIME, (*client).clone().run_polling_updates());
```

Comment:
Can't we use `spawn()`?

```suggestion
        let _handle = CAC_RUNTIME.spawn((*client).clone().run_polling_updates());
```

I think it should be possible, since `run_polling_updates()` does not involve anything that "blocks" the current thread (neither the `tick()` on `tokio`'s `Interval`, nor does the config fetch). So I think it can be moved across threads by the `tokio` scheduler without issues.

Reply:
The function update_cac() is blocking, it tries to acquire a write lock on its config data structure which won't resolve instantly.
https://github.com/juspay/superposition/pull/88/files/2d74390f9b6bd987715434a7ef164a389037316f#diff-0b91e6bae95567932f131f6998310c46af0fb7e3692e85cefeb6aff1a28d6689R140

Reply:
Since we're now using [`tokio::sync::RwLock::write()`](https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html#method.write), it would not block the thread until the lock is acquired, but instead would allow other tasks to be run on the runtime:

> Locks this RwLock with exclusive write access, causing the current task to yield until the lock has been acquired.
>
> The calling task will yield while other writers or readers currently have access to the lock.
>
> Returns an RAII guard which will drop the write access of this RwLock when dropped.
>
> [...]

Reply:
The expectation here is that this function should be run within a thread spawned by the implementing language. Would be better to block the thread so that it does not join back into the main thread